backend_new/src/services/unit-accessibility-priority-type.service.ts

```diff
@@ -0,0 +1,122 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { UnitAccessibilityPriorityType } from '../dtos/unit-accessibility-priority-types/unit-accessibility-priority-type.dto';
+import { UnitAccessibilityPriorityTypeCreate } from '../dtos/unit-accessibility-priority-types/unit-accessibility-priority-type-create.dto';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+import { UnitAccessibilityPriorityTypeUpdate } from '../dtos/unit-accessibility-priority-types/unit-accessibility-priority-type-update.dto';
+
+/*
+  this is the service for unit accessibility priority types
+  it handles all the backend's business logic for reading/writing/deleting unit type data
+*/
+
+@Injectable()
+export class UnitAccessibilityPriorityTypeService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of unit accessibility priority types given the params passed in
+  */
+  async list(): Promise<UnitAccessibilityPriorityType[]> {
+    const rawunitPriortyTypes =
+      await this.prisma.unitAccessibilityPriorityTypes.findMany();
+    return mapTo(UnitAccessibilityPriorityType, rawunitPriortyTypes);
+  }
+
+  /*
+    this will return 1 unit accessibility priority type or error
+  */
+  async findOne(
+    unitAccessibilityPriorityTypeId: string,
+  ): Promise<UnitAccessibilityPriorityType> {
+    const rawunitPriortyTypes =
+      await this.prisma.unitAccessibilityPriorityTypes.findFirst({
+        where: {
+          id: {
+            equals: unitAccessibilityPriorityTypeId,
+          },
+        },
+      });
```

Comment:
I'm ok with either way, but you could also do `findUnique` since this is the primary key

```
      await this.prisma.unitAccessibilityPriorityTypes.findUnique({
        where: {
          id: unitAccessibilityPriorityTypeId
        },
      });

Reply:
updated!